src/data_chain/mod.rs

```diff
@@ -60,11 +60,64 @@ where
 #[cfg(test)]
 mod tests {
     use super::*;
+    use GROUP_SIZE;
     use full_info::FullInfo;
+    use itertools::{self, Itertools};
     use maidsafe_utilities::SeededRng;
+    use routing_table::Prefix;
     use rust_sodium;
+    use serde::Serialize;
     use std::iter;
 
+    /// Creates a collection of `GROUP_SIZE` `FullInfo`s for nodes with ages 5, 6, 7, etc.  If
+    /// `in_prefix` is `Some`, all nodes' IDs will fit that prefix.
+    pub fn create_full_infos(in_prefix: Option<Prefix>) -> Vec<FullInfo> {
+        let mut rng = SeededRng::thread_rng();
+        unwrap!(rust_sodium::init_with_rng(&mut rng));
+
+        let initialiser = || if let Some(prefix) = in_prefix {
+            FullInfo::within_range(0, &prefix.lower_bound(), &prefix.upper_bound())
+        } else {
+            FullInfo::node_new(0)
+        };
+
+        let mut nodes = itertools::repeat_call(initialiser)
+            .take(GROUP_SIZE)
+            .collect_vec();
+        nodes.sort_by(|lhs, rhs| lhs.public_info().cmp(rhs.public_info()));
+        for (index, node) in nodes.iter_mut().enumerate() {
+            node.set_age(index as u8 + 5);
+        }
+        nodes
+    }
+
+    /// Create a block.  If `fill` is true, add votes for every member of `nodes`, otherwise only
+    /// add votes until the block is valid.
```

Comment:
just a note: when `fill` is false, only the first Q nodes will sign to make the block valid.
we may need `Q random nodes to sign` as well, but this can be done later on.